src/Uno.UWP/Extensions/PermissionsHelper.cs

```diff
@@ -40,9 +40,64 @@ public static bool IsDeclaredInManifest(string permission)
 			var requestedPermissions = packageInfo?.RequestedPermissions;
 
 			return requestedPermissions?.Any(r => r.Equals(permission, StringComparison.OrdinalIgnoreCase)) ?? false;
-		}
+		}
+
+		/// <summary>
+		/// Checks if the given Android permissions are declared in manifest file.
+		/// </summary>
+		/// <param name="permission">Array of permissions.</param>
+		/// <returns>true if all permissions are defined</returns>
+		public static bool AreDeclaredInManifest(string[] permissions)
+		{
+			foreach(var permission in permissions)
+			{
+				if (!IsDeclaredInManifest(permission))
+				{
+					return false;
+				}
+			}
+
+			return true;
+		}
+
+		/// <summary>
+		/// Ensures that the given Android permissions are declared in manifest file.
+		/// </summary>
+		/// <param name="permission">Array of permissions.</param>
+		public static void EnsuresPermissionsAreDeclaredInManifest(string[] permissions)
+		{
+			var context = Application.Context;
+			var packageInfo = context.PackageManager.GetPackageInfo(context.PackageName, PackageInfoFlags.Permissions);
+			var requestedPermissions = packageInfo?.RequestedPermissions;
 
+			if(requestedPermissions is null)
+			{
+				throw new UnauthorizedAccessException("no permissions in Manifest defined (no permission at all)");
+			}
+
+			foreach(var permission in permissions)
+			{
+				if (!IsDeclaredInManifest(permission))
+				{
+					throw new UnauthorizedAccessException("no " + permission + " permission in Manifest defined");
+				}
+			}
+
+		}
+
 		/// <summary>
+		/// Validate if a given permission was granted to the app but not request it to the user
+		/// <remarks>
+		/// This should not be invoked directly from the application code.
+		/// You should use the extension methods in <see cref="PermissionsServiceExtensions"/>.
+		/// </remarks>
+		/// </summary>
+		/// <param name="permissionIdentifier">A permission identifier defined in Manifest.Permission.</param>
+		/// <returns>A <see cref="Task"/> representing the asynchronous operation</returns>
+		public static Task<bool> CheckPermission(string permissionIdentifier)
```

Comment:
Let's follow the async naming guidelines:

```suggestion
		public static Task<bool> CheckPermissionAsync(CancellationToken ct, string permissionIdentifier)
```

Reply:
But my idea for this method is shortcut for
` if(!await CheckPermission(CancellationToken.None, "permission") do_something();`
Your changes makes CheckPermissionAsync exactly same as _checkPermission